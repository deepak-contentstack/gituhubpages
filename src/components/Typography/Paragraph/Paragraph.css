@import '../../../../tokens/variables.scss';
@import '../../../../tokens/tokens.css';
@import '../../../static/styles/css/typography.css';
@import '../../../static/styles/css/variable.css';
@import '../../../static/styles/css/common.css';

p {
    color: $color-font-base;
    font-family: $font-family-primary;
    font-size: $size-font-medium;
    font-weight: $font-weight-regular;
}