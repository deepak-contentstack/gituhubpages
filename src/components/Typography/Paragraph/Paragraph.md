## **Paragraph**

Paragraphs let you add regular text.

## When to use 

Paragraphs let you add descriptive text that relate content to topics. They should not be too long or too short.

## Properties

### `text`
- Data that you want to display in tag

### `testId`
- Pass an ID that you can use for testing purposes. It is applied as a data attribute (data-test-id)

### `className`
- Pass the class names to be appended to this prop

### `tagName`
- Name of the <html> tag