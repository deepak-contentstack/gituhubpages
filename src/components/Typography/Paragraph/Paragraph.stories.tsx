import React from 'react'
import DocSubData from '../../../utils/components/DocSubData'
import Paragraph from '../Paragraph/Paragraph'
import notes from './Paragraph.md'

const componentDescription = 'Paragraphs let you add regular text.'

const usageDescription = 'Paragraphs let you add descriptive text that relate content to topics. They should not be too long or too short.'

export default {
    title: 'Components / Typography/Paragraph',
    component: Paragraph,
    argTypes: {
        className: {
            control: 'disabled'
        },
        tagName: {
            control: 'disabled'
        },
    },
    parameters: {
        componentSubtitle: <DocSubData componentDescription={componentDescription} usageDescription={usageDescription} />,
        design: {
            type: 'figma',
            url: 'https://www.figma.com/file/ovlEsic8nr81y5BOu1OlqD/ui-inventory-v2?node-id=22%3A2',
        }
    },

}

export const Default = (args) => (
    <Paragraph tagName='p' text={args.text} />
)
Default.args = {
    text: 'Paragraph text'
}
