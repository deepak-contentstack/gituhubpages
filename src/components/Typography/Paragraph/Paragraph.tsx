import React, { Component } from 'react'
import './Paragraph.css'

export type ParagraphProps = {
  /**
  * Pass the class names to be appended to this prop
  */
  className?: string
  /**
  * Name of the html tag
  */
  tagName?: 'p'
  /**
  * Data that you want to display in tag
  */
  text?: string
  /**
  * Pass an ID that you can use for testing purposes. It is applied as a data attribute (data-test-id)
  */
  testId?: string;
} & typeof defaultProps

const defaultProps = {
  text: 'Paragraph',
  testId: 'cs-paragraph-tag',
  tagName: 'p'
}

export class Paragraph extends Component<ParagraphProps> {
  static defaultProps = defaultProps

  render() {
    const { tagName, text, className, testId } = this.props
    const Element: any = tagName
    return (
      <Element className={className} data-test-id={testId}>
        {text}
      </Element>
    )
  }
}

export default Paragraph
