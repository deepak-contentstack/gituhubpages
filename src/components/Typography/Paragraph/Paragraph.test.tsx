import React from 'react'
import renderer from 'react-test-renderer'
import Paragraph from '../Paragraph/Paragraph'

describe('Paragraph', () => {
  it('Paragraph tag', () => {
    const DefaultParagraphTag = () => {
      return (
        <Paragraph tagName='p' text='Paragraph' />
      )
    }
    const tree = renderer.create(<DefaultParagraphTag />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders the component with an additional class name', () => {
    const DefaultParagraphTag = () => {
      return (
        <Paragraph tagName='p' className="paragraph-tag" text='Paragraph' />
      )
    }
    const tree = renderer.create(<DefaultParagraphTag />).toJSON();
    expect(tree.props.className.includes("paragraph-tag")).toBe(true);
  })

})
