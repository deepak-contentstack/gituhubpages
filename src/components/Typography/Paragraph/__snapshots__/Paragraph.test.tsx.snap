// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph Paragraph tag 1`] = `
<p
  data-test-id="cs-paragraph-tag"
>
  Paragraph
</p>
`;
